src/file.c

```diff
@@ -28,17 +33,25 @@ int FileTerm(File *file)
     int fd = file->fd;
 
     if (fd == -1) {
-        return RR_OK;
+        rc = RR_OK;
+        goto out;
     }
 
     rc = FileFlush(file);
-    file->fd = -1;
 
     if (close(fd) != 0) {
         LOG_WARNING("error, fd: %d, close(): %s", file->fd, strerror(errno));
-        return RR_ERROR;
+        rc = RR_ERROR;
+        goto out;
     }
 
+out:
+    RedisModule_Free(file->buf);
+    *file = (File){
+        .fd = -1,
+        .buf = NULL,
```

Comment:
`wpos` will continue to point to an invalid address.

Reply:
(File) { .fd = -1} syntax sets rest of the struct variables to zero. So, other pointers will be NULL. `.buf = NULL` was also unnecessary here but I guess l left it for documentation purposes (easier to see on a quick look).

Reply:
right, missed the assignment here to a new obj...